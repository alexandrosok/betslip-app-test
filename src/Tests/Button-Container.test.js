import React from 'react';
import {shallow, configure} from 'enzyme';
import ButtonContainer from "../components/Button-Container/button-container";
import Adapter from 'enzyme-adapter-react-16'
import {shallowToJson} from 'enzyme-to-json';

configure({adapter: new Adapter()})

describe('ButtonContainer', () => {
    const wrapper = shallow(<ButtonContainer/>);
    const mockSubmit = jest.fn();

    it('should render correctly', () => {
        const output = shallow(
            <ButtonContainer eventId="mockEventId"/>
        );
        expect(shallowToJson(output)).toMatchSnapshot();
    });

    test('ButtonContainer component elements render', () => {
        expect(wrapper.find('.place-bet-button')).toHaveLength(1);
    });

    it('should render path element with the expected href attribute', () => {
        const button = wrapper.find('.place-bet-button');
        expect(button.props()).toHaveProperty('href', '#');
    });

    it('it calls the placeBet method', () => {
        wrapper.instance().placeBet = mockSubmit;
        wrapper.instance().placeBet();
        expect(mockSubmit).toHaveBeenCalled();
    });

});
