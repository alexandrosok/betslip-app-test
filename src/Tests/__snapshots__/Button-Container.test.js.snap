// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should render correctly 1`] = `
<div
  className="bet-button-container"
>
  <a
    className="place-bet-button"
    href="#"
    onClick={[Function]}
  >
    <span>
      Place A Bet
    </span>
  </a>
</div>
`;
